libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2164,6 +2164,52 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 			m_errorReporter.typeError(errorId, paramArgMap[i]->location(), description);
 		}
 	}
+
+	TypePointers const& returnParameterTypes = _functionType->returnParameterTypes();
+	bool isLibraryCall = (_functionType->kind() == FunctionType::Kind::DelegateCall);
+	bool callRequiresABIEncoding =
+		// ABIEncode/ABIDecode calls not included because they should have been already validated
+		// at this point and they have variadic arguments so they need special handling.
+		_functionType->kind() == FunctionType::Kind::DelegateCall ||
+		_functionType->kind() == FunctionType::Kind::External ||
+		_functionType->kind() == FunctionType::Kind::Creation ||
+		_functionType->kind() == FunctionType::Kind::Event;
+
+	if (callRequiresABIEncoding && !experimentalFeatureActive(ExperimentalFeature::ABIEncoderV2))
+	{
+		solAssert(!isVariadic, "");
+		solAssert(parameterTypes.size() == arguments.size(), "");
+		solAssert(!_functionType->isBareCall(), "");
+		solAssert(*_functionCall.annotation().kind == FunctionCallKind::FunctionCall, "");
+
+		for (size_t i = 0; i < arguments.size(); ++i)
+		{
+			solAssert(parameterTypes[i], "");
```

Comment:
Does this work for named parameters?

Reply:
I think it does because I'm checking parameters of the function type, not arguments actually passed to the call. The assumption is that they will be converted to the required parameter types anyway.

But I'll add a test for this just to be safe.

Reply:
You're right, it's broken. The check itself works but the error location indicates the wrong argument.

Reply:
Done. Problem fixed and a new test added.